src/config/field-overrides/dataSet.js

```diff
@@ -1,7 +1,98 @@
+import React from 'react';
 import OrganisationUnitTreeMultiSelect from '../../forms/form-fields/orgunit-tree-multi-select';
 import DataSetElementField from './data-set/DataSetElementField.component';
 import DataInputPeriods from './data-set/DataInputPeriods.component';
 import PeriodTypeDropDown from '../../forms/form-fields/period-type-drop-down';
+import Checkbox from '../../forms/form-fields/check-box';
+import {RadioButton, RadioButtonGroup} from "material-ui/RadioButton";
+import log from "loglevel";
+
+class CustomTestComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            renderAsTabs: props.value,
+            displayOptions: this.parseDisplayOptions()
+        };
+
+    }
+
+    parseDisplayOptions = () =>  {
+        try {
+            return this.props
+                && this.props.model['displayOptions']
+                && JSON.parse(this.props.model['displayOptions'])
+        } catch (e) {
+            log.error(e);
+            return undefined
+        }
+    }
+
+    updateTabsDirection = (tabsDirection) => {
+        const newDisplayOptions = {
+            ...this.state.displayOptions,
+            tabsDirection
+        }
+        this.setState((prevState, _) => ({
+            ...prevState,
+            displayOptions: newDisplayOptions
+        }));
+        this.props.model.displayOptions = JSON.stringify(newDisplayOptions)
+    }
+
+    onDisplayOptionsChanged = (event) =>  {
+        const tabsDirection = event.target.value
+        this.updateTabsDirection(tabsDirection)
+
+    }
+
+    onDisplayAsTabsChanged = (event) =>  {
```

Comment:
Minor: could we rename this to `onRenderAsTabsChanged` to be consistent with the name of the field/property?

Reply:
absolutely